spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -194,21 +172,64 @@ public RabbitConsumerProperties getExtendedConsumerProperties(String channelName
 	}
 
-
 	@Override
 	public RabbitProducerProperties getExtendedProducerProperties(String channelName) {
 		return this.extendedBindingProperties.getExtendedProducerProperties(channelName);
 	}
 
 	@Override
-	protected MessageProducer createConsumerEndpoint(String name, String group, Queue destination,
-			ExtendedConsumerProperties<RabbitConsumerProperties> properties) {
+	protected MessageHandler createProducerMessageHandler(final ProducerDestination producerDestination,
+														ExtendedProducerProperties<RabbitProducerProperties> producerProperties)
+			throws Exception {
+		String prefix = producerProperties.getExtension().getPrefix();
+		String exchangeName = producerDestination.getName();
+		String destination = StringUtils.isEmpty(prefix) ? exchangeName : exchangeName.substring(prefix.length());
+		final AmqpOutboundEndpoint endpoint = new AmqpOutboundEndpoint(buildRabbitTemplate(producerProperties.getExtension()));
+		endpoint.setExchangeName(producerDestination.getName());
+		RabbitProducerProperties extendedProperties = producerProperties.getExtension();
+		String routingKeyExpression = extendedProperties.getRoutingKeyExpression();
+		if (!producerProperties.isPartitioned()) {
+			if (routingKeyExpression == null) {
+				endpoint.setRoutingKey(destination);
+			}
+			else {
+				endpoint.setRoutingKeyExpressionString(routingKeyExpression);
+			}
+		}
+		else {
+			if (routingKeyExpression == null) {
+				endpoint.setRoutingKeyExpressionString(buildPartitionRoutingExpression(destination));
+			}
+			else {
+				endpoint.setRoutingKeyExpressionString(buildPartitionRoutingExpression(routingKeyExpression));
+			}
+		}
+		if (extendedProperties.getDelayExpression() != null) {
+			endpoint.setDelayExpressionString(extendedProperties.getDelayExpression());
+		}
+		DefaultAmqpHeaderMapper mapper = DefaultAmqpHeaderMapper.outboundMapper();
+		mapper.setRequestHeaderNames(extendedProperties.getRequestHeaderPatterns());
+		mapper.setReplyHeaderNames(extendedProperties.getReplyHeaderPatterns());
+		endpoint.setHeaderMapper(mapper);
+		endpoint.setDefaultDeliveryMode(extendedProperties.getDeliveryMode());
+		endpoint.setBeanFactory(this.getBeanFactory());
+		endpoint.afterPropertiesSet();
+		return endpoint;
+	}
+
+	@Override
+	protected MessageProducer createConsumerEndpoint(ConsumerDestination consumerDestination, String group,
+													ExtendedConsumerProperties<RabbitConsumerProperties> properties) {
 
 		DirectChannel convertingBridgeChannel = new DirectChannel();
 		convertingBridgeChannel.setBeanFactory(this.getBeanFactory());
-		final String baseQueueName = baseQueueName(name, group);
-		convertingBridgeChannel.setBeanName(baseQueueName + ".bridge");
 
+		String prefix = properties.getExtension().getPrefix();
+		String destination = consumerDestination.getName();
+		String prefixStripped = StringUtils.isEmpty(prefix) ? destination : destination.substring(prefix.length());
```

Comment:
`destination.substring(prefix.length())` ???
But who said that all destinations will be with the prefix?
```
destination.startsWith(prefix)
```
first of all?

What am I missing?
Thanks